timeshim/time.go

```diff
@@ -12,22 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package time
+package timeshim
 
 import (
 	"time"
 )
 
 // Time is our shim interface to the time package.
-type Time interface {
+type Interface interface {
 	Now() time.Time
 	Since(t time.Time) time.Duration
 	Until(t time.Time) time.Duration
 	After(t time.Duration) <-chan time.Time
 }
 
-func NewRealTime() Time {
-	return &realTime{}
```

Comment:
Just wondering is this an error to return a pointer as `Time` interface where the `realTime` structure's method only has value receivers. Am I correct that `time.NewRealTime().Now()` will return compile error and `(*time.NewRealTime()).Now()` will not.

Reply:
No, it worked fine before; you can call a non-pointer receiver on a pointer but it has copy semantics so side effects to the struct would be lost.  (This struct is 0-size so no issue here either way.)